.github/workflows/rust-build.yml

```diff
@@ -21,7 +20,10 @@ jobs:
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
+          # scope the cache by branch s.t. it doesn't fall back on base branch if there's a cache miss
+          # see: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+          key:
+            ${{ runner.os }}-cargo-${{ github.ref_name }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
The force update I had in mind was an update with the same key. I already saw that it is not possible. Let's try to use some approach for matching keys like u suggested
ref: https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache

The one in this commit does not look into the main cache (it should)